apps/total-typescript/src/server/routers/progress.ts

```diff
@@ -2,39 +2,60 @@ import {createRouter} from 'server/createRouter'
 import {z} from 'zod'
 import {getSdk} from '@skillrecordings/database'
 import {SubscriberSchema} from 'schemas/subscriber'
+import {getToken} from 'next-auth/jwt'
 
-export const progressRouter = createRouter().mutation('add', {
-  input: z.object({
-    lessonSlug: z.string(),
-  }),
-  async resolve({ctx, input}) {
-    const {findOrCreateUser, completeLessonProgressForUser} = getSdk()
-    try {
-      const subscriberCookie = ctx.req.cookies['ck_subscriber']
+export const progressRouter = createRouter()
+  .mutation('add', {
+    input: z.object({
+      lessonSlug: z.string(),
+    }),
+    async resolve({ctx, input}) {
+      const {findOrCreateUser, completeLessonProgressForUser} = getSdk()
+      try {
+        const subscriberCookie = ctx.req.cookies['ck_subscriber']
 
-      if (!subscriberCookie) {
-        console.debug('no subscriber cookie')
-        return {error: 'no subscriber found'}
-      }
+        if (!subscriberCookie) {
```

Comment:
⚠️  I've noticed that progress tracking won't currently work for users that haven't subscribed to CK but have an account with purchase...